irc_client/views.py

```diff
@@ -84,10 +85,33 @@ def __init__(self, irc_widget: IrcWidget, *, parent_view_id: str = ""):
 
         self.log_file: IO[str] | None = None
 
-    def stop_logging(self) -> None:
+    def get_log_name(self) -> str | None:
+        return None
+
+    def close_log_file(self) -> None:
         if self.log_file is not None:
             print("*** LOGGING ENDS", time.asctime(), file=self.log_file, flush=True)
             self.log_file.close()
+            self.log_file = None
+
+    def open_log_file(self) -> None:
+        self.close_log_file()
+
+        name = self.get_log_name()
+        if name is None:
+            return
+        name = re.sub(r"[^A-Za-z0-9-_#]", "", name)
```

Comment:
A better solution might be to allow only the characters that are valid in nicknames and in Windows file names. It might also be good to replace them with `_`, so that nicknames `||` and `|||` get different log files.

Reply:
nah, whitelist is better